### ISSUE TITLE
Conditional names and descriptions

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#60196]
### Is your feature request related to a problem? Please describe.

Since I added wild carrots to the list of flowers that can spawn, your harvests will not be tainted anymore by the `FORAGE_POISON` flag. The reason it has the flag is because wild carrots (also called queen anne's lace: there is a tiny bit of red in the center of the flower) look very similar to poison hemlock.

### Solution you would like.

I'd love having conditional descriptions and item names. This way you could turn both poison hemlock (does not exist ingame yet) and wild carrot plants into a "white flower" with an identical description. Same applies for the harvested items.

This could have some great potential if it could make use of the conditionals system. I would imagine the most common uses would be basing stuff on skill level (e.g survival 0: "white flower", survival 1: "poison hemlock"), stats (e.g int 3: "blue sink juice", int 10: "allpurpose cleaner") or profiencies (e.g default: "shoulder guard", with armor smithing proficiency: "pauldron"). Maybe even traits (e.g default: "mannequin", as churl: "marionette of the devil").

Or if you have crossed the elf-a threshold, some mycus terrain/furniture may have a hint like "your elf senses tell you to kill it". There are so many possibilities for great details here.

### Describe alternatives you have considered.

Dealing with my specific carrot/poison hemlock thing by adding a poison hemlock flower, doing copy-from and hardcode `HIDDENPOISON` into the harvested item. Hacky.

**Comment by ANickelN** [comment]
I don't know much about the feasibility of this code-wise, but I love the idea and the scenarios you presented where it could be used!

**Comment by GuardianDll** [comment]
yeah, it can be used almost everywhere: "generic name" of gun if your range stat is low, monster's with different names, when you met them first time and after a while (e.g. fat zombie - boomer)
